alchemy-core/src/main/java/io/rtr/alchemy/models/Treatment.java

```diff
@@ -31,6 +35,13 @@ public void setDescription(String description) {
         this.description = description;
     }
 
+    private static void validateName(String name) {
+        // see https://stackoverflow.com/questions/1969232/allowed-characters-in-cookies/45536811#45536811
+        if (! name.matches("^[A-Za-z0-9!$&'()*+-.:@_~]*$")) {
```

Comment:
I wouldn't make the name specifically tied to allowable characters for cookies, that's just an implementation detail for why it's problematic for us.  I would adopt a simple pattern, maybe just letters, numbers, underscore and hyphen, nothing else `[A-Za-z0-9-_]`, since these are just meant to be canonical identifiers, not anything more descriptive than that.

Reply:
ok